extensions/big-data-cluster/src/bigDataCluster/tree/controllerTreeNode.ts

```diff
@@ -0,0 +1,376 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import * as vscode from 'vscode';
+import * as azdata from 'azdata';
+import * as nls from 'vscode-nls';
+import { IControllerTreeChangeHandler } from './controllerTreeChangeHandler';
+import { TreeNode } from './treeNode';
+import { IconPath, BdcItemType } from '../constants';
+import { ClusterController, IEndPoint, IControllerError } from '../controller/clusterController';
+
+const localize = nls.loadMessageBundle();
+
+export type ControllerTreeNodeArg = {
+	label: string;
+	treeChangeHandler: IControllerTreeChangeHandler;
+	parent?: ControllerTreeNode;
+	description?: string;
+	nodeType?: string;
+	iconPath?: { dark: string, light: string };
+};
+
+export abstract class ControllerTreeNode extends TreeNode {
+	private _description: string;
+	private _nodeType: string;
+	private _iconPath: { dark: string, light: string };
+	private _treeChangeHandler: IControllerTreeChangeHandler;
+
+	constructor(arg?: ControllerTreeNodeArg) {
```

Comment:
2 issues:
- arg is optional but you reference its fields. These are all potential null references
- I mentioned in the last review that the coding standard is to do things as follows:

```ts
constructor(label: string,
    private _treeChangeHandler: IControllerTreeChangeHandler,
    private _nodeType: string
    private _iconPath: { dark: string, light: string },
    description?: string,
    parent?: TreeNode
) {
    super(label, parent);
    this._description =description || label;
}
```

This is the standard across the codebase and reduces the number of lines of code quite a lot (since most private fields are defined & initialized in 1 line in the constructor

Reply:
fixed.